src/ui/public/directives/config.js

```diff
@@ -8,75 +7,110 @@ var module = uiModules.get('kibana');
 
 
 /**
- * config directive
+ * kbnTopNav directive
  *
- * Creates a full width horizonal config section, usually under a nav/subnav.
+ * The top section that shows the timepicker, load, share and save dialogues.
  * ```
- * <config config-template="configTemplate" config-object="configurable"></config>
+ * <kbn-top-nav name="current-app-for-extensions" config="path.to.menuItems"></kbn-top-nav>
  * ```
  */
 
-module.directive('config', function ($compile) {
+module.directive('kbnTopNav', function (Private) {
+  const filterTemplate = require('ui/chrome/config/filter.html');
+  const intervalTemplate = require('ui/chrome/config/interval.html');
+  function optionsNormalizer(defaultFunction, opt) {
+    if (!opt.key) {
+      return false;
+    }
+    return _.assign({
+      label: _.capitalize(opt.key),
+      hasFunction: !!opt.run,
+      description: ('Toggle ' + opt.key),
+      run: defaultFunction
+    }, opt);
+  }
+  function getTemplatesMap(configs) {
+    const templateMap = {};
+    configs.forEach(conf => {
+      if (conf.template) {
+        templateMap[conf.key] = conf.template;
+      }
+    });
+    return templateMap;
+  }
   return {
     restrict: 'E',
-    scope: {
-      configTemplate: '=',
-      configClose: '=',
-      configSubmit: '=',
-      configObject: '='
+    transclude: true,
+    template: function ($el, $attrs) {
+      // This is ugly
+      // This is necessary because of navbar-extensions
+      // It will no accept any programatic way of setting its name
+      // besides this because it happens so early in the digest cycle
+      return `
+      <navbar class="kibana-nav-options">
+        <div ng-transclude></div>
+        <div class="button-group kibana-nav-actions" role="toolbar">
+          <button
+            ng-repeat="menuItem in kbnTopNav.menuItems"
+            aria-label="{{::menuItem.description}}"
+            aria-haspopup="{{!menuItem.hasFunction}}"
+            aria-expanded="{{kbnTopNav.is(menuItem.key)}}"
+            ng-class="{active: kbnTopNav.is(menuItem.key)}"
+            ng-click="menuItem.run(menuItem)"
+            ng-bind="menuItem.label">
+          </button>
+          <navbar-extensions name="${$attrs.name}"></navbar-extensions>
```

Comment:
You mean ike you can't pu your plugin before the default ones anymore? This is true.